src/plugins/remark-collapse.ts

```
@@ -0,0 +1,124 @@
+          return true; // Remove this node
+        }
+        return false; // Keep other nodes
+      });
+      
+      // The remaining children of the directiveNode will become the content of the <details>
+      // These are Content[] from mdast, which might include block elements.
+      // The `h` function expects PhrasingContent[] for its children if strictly typed.
+      // This implies that the `div` wrapper for content might need to handle FlowContent,
+      // or the `h` function needs to be more flexible or a different node type used.
+      // Transform Content[] into PhrasingContent[] to ensure type safety.
+      const contentChildren = transformToPhrasingContent(directiveNode.children);
```

ESLint

[error] 'transformToPhrasingContent' is not defined.

(no-undef)